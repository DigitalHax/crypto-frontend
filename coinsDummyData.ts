import { CoinDetailed } from '@/types';

export const coinsDummyData: CoinDetailed[] = [
  {
    id: '01coin',
    symbol: 'zoc',
    name: '01coin',
    image:
      'https://assets.coingecko.com/coins/images/5720/large/F1nTlw9I_400x400.jpg?1547041588',
    current_price: 0.00042911,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 0.098696,
    high_24h: 0.00043096,
    low_24h: 0.00041807,
    price_change_24h: 0.00001073,
    price_change_percentage_24h: 2.56431,
    market_cap_change_24h: 0,
    market_cap_change_percentage_24h: 0,
    circulating_supply: 0,
    total_supply: 65658824,
    max_supply: null,
    ath: 0.04813117,
    ath_change_percentage: -99.10953,
    ath_date: '2018-10-10T17:27:38.034Z',
    atl: 0.0003688,
    atl_change_percentage: 16.21278,
    atl_date: '2022-05-12T13:58:36.971Z',
    roi: null,
    last_updated: '2022-05-23T06:48:28.322Z'
  },
  {
    id: '0-5x-long-algorand-token',
    symbol: 'algohalf',
    name: '0.5X Long Algorand Token',
    image:
      'https://assets.coingecko.com/coins/images/12009/large/683JEXMN_400x400.png?1596692452',
    current_price: 16056.05,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 1.61,
    high_24h: 16106.59,
    low_24h: 16028.65,
    price_change_24h: 0.388187,
    price_change_percentage_24h: 0.00242,
    market_cap_change_24h: 0,
    market_cap_change_percentage_24h: 0,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 35644,
    ath_change_percentage: -54.955,
    ath_date: '2021-09-13T00:44:05.420Z',
    atl: 9917.35,
    atl_change_percentage: 61.89858,
    atl_date: '2020-11-05T21:49:34.570Z',
    roi: null,
    last_updated: '2022-05-22T17:17:52.745Z'
  },
  {
    id: '0-5x-long-altcoin-index-token',
    symbol: 'althalf',
    name: '0.5X Long Altcoin Index Token',
    image:
      'https://assets.coingecko.com/coins/images/12008/large/683JEXMN_400x400.png?1596692307',
    current_price: 22202,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 3.33,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 35745,
    ath_change_percentage: -37.88859,
    ath_date: '2021-11-09T16:44:33.671Z',
    atl: 10325,
    atl_change_percentage: 115.02899,
    atl_date: '2020-11-05T21:49:14.354Z',
    roi: null,
    last_updated: '2022-05-12T12:05:35.382Z'
  },
  {
    id: '0-5x-long-ascendex-token-token',
    symbol: 'asdhalf',
    name: '0.5X Long AscendEx Token Token',
    image:
      'https://assets.coingecko.com/coins/images/12003/large/683JEXMN_400x400.png?1596691634',
    current_price: 20838,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 56.26,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 41773,
    ath_change_percentage: -50.11554,
    ath_date: '2022-01-29T10:00:57.004Z',
    atl: 9069.12,
    atl_change_percentage: 129.77201,
    atl_date: '2020-11-05T21:49:33.554Z',
    roi: null,
    last_updated: '2022-04-28T00:04:27.397Z'
  },
  {
    id: '0-5x-long-bitcoin-cash-token',
    symbol: 'bchhalf',
    name: '0.5X Long Bitcoin Cash Token',
    image:
      'https://assets.coingecko.com/coins/images/12006/large/683JEXMN_400x400.png?1596692054',
    current_price: 9055.33,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 85.03,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 20983,
    ath_change_percentage: -56.84366,
    ath_date: '2021-05-06T19:57:55.584Z',
    atl: 8469.94,
    atl_change_percentage: 6.91139,
    atl_date: '2020-10-09T06:04:28.636Z',
    roi: null,
    last_updated: '2022-05-19T23:20:09.313Z'
  },
  {
    id: '0-5x-long-bitcoin-token',
    symbol: 'half',
    name: '0.5X Long Bitcoin Token',
    image:
      'https://assets.coingecko.com/coins/images/11996/large/683JEXMN_400x400.png?1596689733',
    current_price: 19482.61,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 1408.98,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 27469,
    ath_change_percentage: -29.07411,
    ath_date: '2021-11-10T14:19:45.099Z',
    atl: 11222.77,
    atl_change_percentage: 73.59894,
    atl_date: '2020-08-07T18:38:17.669Z',
    roi: null,
    last_updated: '2022-05-21T05:09:24.359Z'
  },
  {
    id: '0-5x-long-cardano-token',
    symbol: 'adahalf',
    name: '0.5X Long Cardano Token',
    image:
      'https://assets.coingecko.com/coins/images/12011/large/683JEXMN_400x400.png?1596692577',
    current_price: 43310,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 1.73,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 95145,
    ath_change_percentage: -54.48004,
    ath_date: '2021-09-02T05:45:49.890Z',
    atl: 15146.08,
    atl_change_percentage: 185.9496,
    atl_date: '2020-11-05T21:49:12.921Z',
    roi: null,
    last_updated: '2022-05-21T20:51:15.560Z'
  },
  {
    id: '0-5x-long-chainlink-token',
    symbol: 'linkhalf',
    name: '0.5X Long Chainlink Token',
    image:
      'https://assets.coingecko.com/coins/images/11992/large/683JEXMN_400x400.png?1596688614',
    current_price: 15759.85,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 2.84,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 34107,
    ath_change_percentage: -53.79358,
    ath_date: '2021-11-10T21:34:13.245Z',
    atl: 15561.78,
    atl_change_percentage: 1.27283,
    atl_date: '2022-05-12T08:38:09.705Z',
    roi: null,
    last_updated: '2022-05-13T12:02:39.630Z'
  },
  {
    id: '0-5x-long-cosmos-token',
    symbol: 'atomhalf',
    name: '0.5X Long Cosmos Token',
    image:
      'https://assets.coingecko.com/coins/images/12007/large/683JEXMN_400x400.png?1596692188',
    current_price: 25079,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 3797.15,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 43691,
    ath_change_percentage: -42.59968,
    ath_date: '2022-01-07T11:46:17.711Z',
    atl: 11274.78,
    atl_change_percentage: 122.43082,
    atl_date: '2020-12-25T02:05:48.106Z',
    roi: null,
    last_updated: '2022-05-16T02:26:23.671Z'
  },
  {
    id: '0-5x-long-dogecoin-token',
    symbol: 'dogehalf',
    name: '0.5X Long Dogecoin Token',
    image:
      'https://assets.coingecko.com/coins/images/12002/large/683JEXMN_400x400.png?1596691478',
    current_price: 121951,
    market_cap: 0,
    market_cap_rank: null,
    fully_diluted_valuation: null,
    total_volume: 28.05,
    high_24h: null,
    low_24h: null,
    price_change_24h: null,
    price_change_percentage_24h: null,
    market_cap_change_24h: null,
    market_cap_change_percentage_24h: null,
    circulating_supply: 0,
    total_supply: null,
    max_supply: null,
    ath: 276270,
    ath_change_percentage: -55.85792,
    ath_date: '2021-05-08T13:28:31.150Z',
    atl: 9832.73,
    atl_change_percentage: 1140.25842,
    atl_date: '2020-11-05T21:44:32.420Z',
    roi: null,
    last_updated: '2022-05-21T04:32:38.284Z'
  }
];
