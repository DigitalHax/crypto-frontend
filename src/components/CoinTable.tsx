import { CoinDetailed } from '@/types';
import { numberWithAbbrev, numberWithCommas } from '@/utils/format';

export const CoinTable = ({
  coinsDetailed
}: {
  coinsDetailed: CoinDetailed[];
}) => {
  return (
    <table className="table-auto w-full mb-px">
      <thead className="text-sm font-semibold uppercase text-slate-500 bg-slate-50 rounded-md">
        <tr>
          <th className="px-2 first:pl-5 last:pr-5 py-3 whitespace-nowrap">
            <div className="font-semibold text-left">Coin</div>
          </th>
          <th className="px-2 first:pl-5 last:pr-5 py-3 whitespace-nowrap">
            <div className="font-semibold text-left">Price (AUD)</div>
          </th>
          <th className="px-2 first:pl-5 last:pr-5 py-3 whitespace-nowrap">
            <div className="font-semibold text-left">24h Change</div>
          </th>
          <th className="px-2 first:pl-5 last:pr-5 py-3 whitespace-nowrap">
            <div className="font-semibold text-left">Volume</div>
          </th>
          <th className="px-2 first:pl-5 last:pr-5 py-3 whitespace-nowrap">
            <div className="font-semibold text-left">Mkt Cap</div>
          </th>
        </tr>
      </thead>
      <tbody className="text-sm divide-y divide-slate-200">
        {coinsDetailed.map((coinDetailed: CoinDetailed, index: React.Key) => (
          <CoinTableRow key={index} coinDetailed={coinDetailed} />
        ))}
      </tbody>
    </table>
  );
};

export const CoinTableRow = ({
  coinDetailed
}: {
  coinDetailed: CoinDetailed;
}) => {
  return (
    <tr>
      <td className="px-2 text-lg first:pl-5 last:pr-5 py-2.5 whitespace-nowrap">
        <div className="flex items-center">
          <img className="inline-flex mr-3 h-8 w-8" src={coinDetailed.image} />
          <div className="text-sm text-slate-800 flex flex-col">
            <span className="font-semibold text-slate-800 inline-flex items-center gap-x-2">
              {coinDetailed.name}
            </span>
            <span className="text-xs text-slate-500">
              {coinDetailed.symbol.toUpperCase()}
            </span>
          </div>
        </div>
      </td>
      <td className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap">
        <span className="text-left font-medium">
          {coinDetailed.current_price
            ? '$' + numberWithCommas(coinDetailed.current_price)
            : '-'}
        </span>
      </td>
      <td className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap">
        <span className="text-left font-medium">
          {coinDetailed.price_change_percentage_24h
            ? coinDetailed.price_change_percentage_24h.toFixed(2) + '%'
            : '-'}
        </span>
      </td>
      <td className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap">
        <span className="text-left font-medium">
          {coinDetailed.total_volume
            ? '$' + numberWithCommas(coinDetailed.total_volume)
            : '-'}
        </span>
      </td>
      <td className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap">
        <span className="text-left font-medium">
          {coinDetailed.market_cap
            ? '$' + numberWithAbbrev(coinDetailed.market_cap)
            : '-'}
        </span>
      </td>
    </tr>
  );
};
