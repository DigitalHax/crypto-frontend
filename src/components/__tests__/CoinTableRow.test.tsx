import { coinsDummyData } from '../../../coinsDummyData';
import renderer from 'react-test-renderer';
import { CoinTableRow } from '../CoinTable';

it('renders correctly', () => {
  const tree = renderer
    .create(<CoinTableRow coinDetailed={coinsDummyData[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
