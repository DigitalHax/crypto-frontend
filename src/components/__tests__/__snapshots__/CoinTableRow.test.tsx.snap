// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td
    className="px-2 text-lg first:pl-5 last:pr-5 py-2.5 whitespace-nowrap"
  >
    <div
      className="flex items-center"
    >
      <img
        className="inline-flex mr-3 h-8 w-8"
        src="https://assets.coingecko.com/coins/images/5720/large/F1nTlw9I_400x400.jpg?1547041588"
      />
      <div
        className="text-sm text-slate-800 flex flex-col"
      >
        <span
          className="font-semibold text-slate-800 inline-flex items-center gap-x-2"
        >
          01coin
        </span>
        <span
          className="text-xs text-slate-500"
        >
          ZOC
        </span>
      </div>
    </div>
  </td>
  <td
    className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap"
  >
    <span
      className="text-left font-medium"
    >
      $0.00042911
    </span>
  </td>
  <td
    className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap"
  >
    <span
      className="text-left font-medium"
    >
      2.56%
    </span>
  </td>
  <td
    className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap"
  >
    <span
      className="text-left font-medium"
    >
      $0.098696
    </span>
  </td>
  <td
    className="px-2 text-sm first:pl-5 last:pr-5 py-3 whitespace-nowrap"
  >
    <span
      className="text-left font-medium"
    >
      -
    </span>
  </td>
</tr>
`;
